nexus/src/app/mod.rs

```diff
@@ -465,6 +473,20 @@ impl Nexus {
     ) -> db::lookup::LookupPath {
         db::lookup::LookupPath::new(opctx, &self.db_datastore)
     }
+
+    pub async fn set_resolver(
+        &self,
+        resolver: internal_dns_client::multiclient::Resolver,
+    ) {
+        *self.resolver.lock().await = resolver;
+    }
+
+    pub async fn resolver(
+        &self,
+    ) -> tokio::sync::MutexGuard<'_, internal_dns_client::multiclient::Resolver>
+    {
+        self.resolver.lock().await
+    }
```

Comment:
I think it's sketchy to expose a whole `MutexGuard` here. If any Nexus methods use the resolver, and we expose this functionality, a caller can do the following:

```rust
let resolver = nexus.resolver().await;
nexus.operation_which_uses_resolver().await; // Now we deadlock
```

Reply:
agreed, fixed in 1bd3ff98